patches/llvm/0002-Add-managarm-target.patch

```diff
@@ -285,13 +288,13 @@ index 000000000..3639920fd
 +  // to the link paths.
 +  path_list &Paths = getFilePaths();
 +
-+  const std::string OSLibDir = getOSLibDir(Triple, Args);
++  const std::string OSLibDir = std::string(getOSLibDir(Triple, Args));
 +  const std::string MultiarchTriple = getMultiarchTriple(D, Triple, SysRoot);
 +
 +  // Add the multilib suffixed paths where they are available.
 +  if (GCCInstallation.isValid()) {
 +    const llvm::Triple &GCCTriple = GCCInstallation.getTriple();
-+    const std::string &LibPath = GCCInstallation.getParentLibPath();
++    const std::string &LibPath = std::string(GCCInstallation.getParentLibPath());
```

Comment:
Are these `std::string` casts also done in toolchains other than Managarm?

Reply:
Yep, verified it with the Linux toolchain and this is what they did. Without the casts it doesn't compile so I presume LLVM changed some internal thing to require this.